src/components/admin/settings-tab.js

```diff
@@ -78,120 +91,114 @@ function SettingsTab({
   const handleSaveButtonClick = async () => {
     const changedSettings = dirtyKeys.reduce((map, key) => ({ ...map, [key]: settings[key] }), {});
     try {
-      await settingsApiClient.saveSettings({ settings: changedSettings });
-      const updatedSettings = { ...lastSavedSettings, ...changedSettings };
-      setLastSavedSettings(updatedSettings);
+      setIsSaving(true);
+      const res = await settingsApiClient.saveSettings({ settings: changedSettings });
```

Comment:
We need to pass `settingsApiClient.saveSettings({ settings: { ...settings, ...changedSettings } })` here, otherwise we lose/clear all other settings but the changed ones 😬

Reply:
Good catch, thanks :grimacing: